codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker.java

```diff
@@ -453,13 +455,33 @@ static int websocketPort(URI wsURL) {
         return wsPort;
     }
 
+    static CharSequence websocketHostValue(URI wsURL) {
+        int port =  wsURL.getPort();
+        if (port == -1) {
+            return wsURL.getHost();
+        }
+        String host = wsURL.getHost();
+        if (port == HttpScheme.HTTP.port()) {
+            return HttpScheme.HTTP.name().contentEquals(wsURL.getScheme())
+                    || "ws".equals(wsURL.getScheme()) ? host : NetUtil.toSocketAddressString(host, port);
+        }
+        if (port == HttpScheme.HTTPS.port()) {
+            return HttpScheme.HTTPS.name().contentEquals(wsURL.getScheme())
+                    || "wss".equals(wsURL.getScheme()) ? host : NetUtil.toSocketAddressString(host, port);
```

Comment:
consider creating a `WebSocketScheme`. These `wss` and `ws` strings are starting to become more plentiful.